src/lib/storage/chunk.hpp

```diff
@@ -181,22 +181,32 @@ class Chunk : private Noncopyable {
   void set_cleanup_commit_id(CommitID cleanup_commit_id);
 
   /**
-   * Executes tasks that are connected with finalizing a chunk. Currently, chunks are made immutable, and
-   * depending on skip_mvcc_check, the MVCC max_begin_cid is set. Finalizing a chunk is the inserter's responsibility.
+   * Executes tasks that are connected with finalizing a chunk. Currently, chunks are made immutable, and if not already
+   * done by Insert operators, the MVCC max_begin_cid is set. Finalizing a chunk is the inserter's responsibility.
    */
   void finalize();
 
+  /**
+   * Insert operators mark that they appended a new chunk to the table. From this moment on, the former last chunk can
+   * be finalized as soon as all pending Inserts commited or rolled back (which can also be immediately). After each
```

Comment:
```c++
/**
 * Insert operators mark that they appended a new chunk to the table. From this moment on, the former last chunk can
 * be finalized as soon as all pending Inserts committed or rolled back. If there are no pending Inserts, i.e., the
 * chunk was filled exactly to its target size and all Inserts are commited/rolled back, the chunk is immediately
 * finalized. Otherwiese, each Insert operator tries to finalize the chunks it inserted data into on commit or
 * rollback. If a chunk is finalizable (i.e., it reached the target size and a new chunk was added to the table) and
 * no Inserts are pending, the chunk is set to be immutable.
 */
```

Reply:
commited

Reply:
Otherwiese

Reply:
"If there are no pending Inserts, i.e., the chunk was filled exactly to its target size...". Remove the "exactly"?
Sentence two and three overlap quite a bit.

Reply:
```c++
/**
 * Insert operators mark that they appended a new chunk to the table. From this moment on, the former last chunk can
 * be finalized as soon as all pending Inserts committed or rolled back. If there are no pending Inserts, i.e., the
 * chunk was filled to its target size and all Inserts are committed/rolled back, the chunk is immediately finalized.
 * Otherwise, each Insert operator tries to finalize the chunks it inserted data into on commit or rollback.
 * Finalizing marks the chunk as immutable.
 */
```

Reply:
`Finalizing marks the chunk as immutable.`

Should we move away from `finalize`? Not really that telling, right?

Reply:
Can you merge sentence 2 and 4 in the comment?